scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -755,43 +773,50 @@ object Execution {
     // if so we remove them from the cache.
     // Anything not already ran we run as part of a single flow def, using their combined counters for the others
     protected def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) = {
-      Trampoline(cache.getOrElseInsert(conf, this, {
-        val cacheLookup: List[(ToWrite[_], Either[Promise[Map[Long, ExecutionCounters]], Future[Map[Long, ExecutionCounters]]])] =
+      lazy val uncachedFutureCancel = {
+        val cacheLookup: List[(ToWrite[_], (Either[CPromise[Map[Long, ExecutionCounters]], CFuture[Map[Long, ExecutionCounters]]]))] =
           (head :: tail).map{ tw => (tw, cache.getOrLock(conf, tw)) }
         val (weDoOperation, someoneElseDoesOperation) = unwrapListEither(cacheLookup)
 
-        val otherResult = failFastSequence(someoneElseDoesOperation.map(_._2))
-        otherResult.value match {
-          case Some(Failure(e)) => Future.failed(e)
-          case _ => // Either successful or not completed yet
-            val localFlowDefCountersFuture: Future[Map[Long, ExecutionCounters]] =
-              weDoOperation match {
-                case all @ (h :: tail) =>
-                  val futCounters: Future[Map[Long, ExecutionCounters]] =
-                    cache.writer
-                      .execute(conf, all.map(_._1))
-                      .map(Map(_))
-
-                  // Complete all of the promises we put into the cache
-                  // with this future counters set
-                  all.foreach {
-                    case (toWrite, promise) =>
-                      promise.completeWith(futCounters)
-                  }
-                  futCounters
-                case Nil =>
-                  // No work to do, provide a fulled set of 0 counters to operate on
-                  Future.successful(Map.empty)
+        val otherResult = CFuture.failFastSequence(someoneElseDoesOperation.map(_._2))
+
+        val localFlowDefCountersFuture: CFuture[Map[Long, ExecutionCounters]] =
```

Comment:
ah, i think you're right. i pulled this out of the `case _ =>` because i thought i needed the cancellation handler. but since we are starting the work for these futures in this method, there isn't anything to cancel in the failed case.